climada/util/api_client.py

```diff
@@ -179,18 +251,45 @@ class AmbiguousResult(Exception):
     class NoResult(Exception):
         """Custom Exception for No Query Result"""
 
-    def __init__(self):
+    class NoConnection(Exception):
+        """To be raised if there is no internet connection and no cached result."""
+
+
+    @staticmethod
+    def _is_online(url):
+        host = [x for x in url.split('/')
+                if x not in ['https:', 'http:', '']][0]
+        port = 80 if host.startswith('http://') else 443
```

Comment:
```suggestion
        port = 80 if url.startswith('http://') else 443
```
i think using `host` here will always return port 80, since the http has been stripped from it